custom-instrumentation/distributed-tracing/consumer.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const newrelic = require('newrelic')
+const { Worker } = require('bullmq')
+const IORedis = require('ioredis')
+
+const connection = new IORedis({
+  maxRetriesPerRequest: null
+})
+
+// since BullMQ is not auto instrumented by the newrelic node agent, we have to manually start a transaction
+return newrelic.startBackgroundTransaction('Message queue - consumer', function innerHandler() {
```

Comment:
so this creates only 1 transaction. you _could_ wrap every consumption in a transaction because right now it's 1. I know in standard messages queues we create 1 transaction for every consumption

Reply:
I would wrap it in the job processor, right?